mountpoint-s3-fs/src/memory/buffers.rs

```diff
@@ -220,4 +227,24 @@ mod tests {
         let bytes = pool_buffer.into_bytes();
         assert_eq!(bytes.as_ref(), &data[..appended_size]);
     }
+
+    #[test_matrix([primary, secondary], [SMALLER_THEN_BUFFER_SIZE, BUFFER_SIZE, LARGER_THAN_BUFFER_SIZE])]
+    fn test_pool_buffer_mut_fill(create_fn: fn(usize) -> PoolBuffer, read_size: usize) {
```

Comment:
Should also check the behaviour when the buffer already has some content in it. Success cases should leave the buffer full, and failures should leave the length unchanged.